integrationTests/chainSimulator/common.go

```diff
@@ -210,3 +219,40 @@ func getEsdtIdentifier(t *testing.T, nodeHandler process.NodeHandler, ticker str
 	require.Fail(t, "could not issue semi fungible")
 	return ""
 }
+
+// Deposit will deposit tokens in a contract
+func Deposit(
+	t *testing.T,
+	cs ChainSimulator,
+	sender []byte,
+	nonce *uint64,
+	contract []byte,
+	tokens []ArgsDepositToken,
+	receiver []byte,
+) {
+	require.True(t, len(tokens) > 0)
+
+	depositArgs := multiEsdtTransfer +
+		"@" + hex.EncodeToString(contract) +
+		"@" + fmt.Sprintf("%02X", len(tokens))
+
+	for _, token := range tokens {
```

Comment:
How does this work?
Shouldn't those fields be encoded using data codec?

Reply:
For tokens we don't need. We need for `TransferData` (gas, function, args) the codec which I forgot to add to this function.
I created a task `MX-15523` to integrate the new version of sdk-abi-go, then I need to add dependence in the main node package then encode transfer data arguments.